### ISSUE TITLE
Dependabot doesn't recognize Gemfile "file:" directive for Ruby version

### ISSUE THREAD

**Issue by bewuethr** [dependabot/dependabot-core#8539]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Package ecosystem

Bundler

### Package manager version

Bundler 2.4.22

### Language version

Ruby 3.2.2

### Manifest location and content before the Dependabot update

/Gemfile, contents:

```ruby
source "https://rubygems.org"

ruby file: ".tool-versions"

gem "jekyll"
gem "jekyll-feed"
gem "jekyll-include-cache"
gem "jekyll-last-modified-at"
gem "jekyll-paginate-v2"
gem "jekyll-redirect-from"
gem "jekyll-remote-theme"
gem "jekyll-sitemap"
gem "jemoji"
```

### dependabot.yml content

```yaml
version: 2

updates:
  # GitHub Actions
  - package-ecosystem: github-actions
    directory: /
    schedule:
      interval: daily
      time: '08:00'  # yamllint disable-line rule:quoted-strings
      timezone: America/Toronto

  # Ruby Gems
  - package-ecosystem: bundler
    directory: /
    schedule:
      interval: daily
      time: '08:00'  # yamllint disable-line rule:quoted-strings
      timezone: America/Toronto
```

### Updated dependency

n/a

### What you expected to see, versus what you actually saw

Dependabot update jobs fail with this message:

```none
Bundler::Dsl::DSLError with message: [!] There was an error parsing `Gemfile`: No such file or directory @ rb_sysopen - dependabot_tmp_dir/.tool-versions. Bundler cannot continue.

 #  from dependabot_tmp_dir/Gemfile:3
 #  -------------------------------------------
 #  
 >  ruby file: ".tool-versions"
 #  
 #  -------------------------------------------
```

### Native package manager behavior

Bundler accesses the specified file `.tool-versions` to extract the Ruby version.

### Images of the diff or a link to the PR, issue, or logs

- Dependabot update log: https://github.com/swissclubto/swissclubto.github.io/network/updates/758615140
- The `ruby file:` option is documented in the [bundler docs](https://bundler.io/v2.4/man/gemfile.5.html#VERSION-required-)
- Support was introduced in [bundler-v2.4.20](https://github.com/rubygems/rubygems/releases/tag/bundler-v2.4.20)
- PR implementing it: https://github.com/rubygems/rubygems/pull/6898

### Smallest manifest that reproduces the issue

`Gemfile`:

```ruby
source "https://rubygems.org"

ruby file: ".tool-versions"
```

`Gemfile.lock`:

```
GEM
  remote: https://rubygems.org/
  specs:

PLATFORMS
  x86_64-linux

DEPENDENCIES

RUBY VERSION
   ruby 3.2.2p53

BUNDLED WITH
   2.4.22
```

`.tool-versions`:

```
ruby 3.2.2
```

**Comment by bewuethr** [comment]
It looks like if this method

https://github.com/dependabot/dependabot-core/blob/ffb64af1ecb08a953520f412afc91e073c930e1e/bundler/lib/dependabot/bundler/file_fetcher.rb#L115-L122

also checked for `.tool-versions`, things might work?

**Comment by deivid-rodriguez** [comment]
Yeah, that sounds like it might work. Would you be willing to open a PR for it?

**Comment by grzuy** [comment]
I suspect this will be fixed by https://github.com/dependabot/dependabot-core/pull/9070

**Comment by bewuethr** [comment]
> I suspect this will be fixed by #9070

Looks like it! Excellent 😃

**Comment by bewuethr** [comment]
I think #9070 fixed this! Thanks @kpumuk 😃
